serializer.h

```diff
@@ -1757,6 +1765,122 @@ void serialize(Archive & archive, const polymorphic_wrapper<Type> & object)
     registry_instance.serialize(archive, *object);
 }
 
+/**
+ * Support types introduced in c++17
+ */
+#if __cplusplus > 201402L
+/**
+ * The base function of the variant serialization helper function.
+ */
+template <typename Archive, typename Variant,
+    typename = typename Archive::loading
+>
+void serialize(Archive &, const Variant &, std::size_t)
+{
+}
+
+/**
+ * Serialize std::variant, operates on loading(input) archives.
+ * Enumerate on all the variant types, until the active type is
+ * reached, and serialize it.
+ */
+template <typename Archive, typename VariantItem, typename... VariantItems,
+    typename Variant, typename = typename Archive::loading
+>
+void serialize(Archive & archive, Variant & variant, std::size_t index)

```

Comment:
That's a clever idea, I think you can do it in O(1) though, if not I'll reserve some time to do that.

Reply:
Couldn't figure out a way to do it in O(1) though....

Reply:
One way to do it is via array access you need to define an array whose size is the highest index of the variant plus 1, then invoke the array at the runtime index. This solution makes an absolute jmp however which may be slower. Either you check it or I can do it during the weekend probably.